public-ui/libproject/devstart.rb

```diff
@@ -0,0 +1,248 @@
+end
+
+Common.register_command({
+  :invocation => "install-dependencies",
+  :description => "Installs dependencies via yarn.",
+  :fn => Proc.new { |*args| install_dependencies(*args) }
+})
+
+def swagger_regen()
+  common = Common.new
+  Workbench::Swagger.download_swagger_codegen_cli
+  common.run_inline %W{docker-compose run --rm public-ui yarn run codegen}
+end
+
+Common.register_command({
+  :invocation => "swagger-regen",
+  :description => "Regenerates API client libraries from Swagger definitions.",
+  :fn => Proc.new { |*args| swagger_regen(*args) }
+})
+
+class BuildOptions
+  # Keep in sync with .angular-cli.json.
+  ENV_CHOICES = %W{local-test local test staging stable}
+  attr_accessor :env
+
+  def initialize
+    self.env = "dev"
+  end
+
+  def parse cmd_name, args
+    parser = OptionParser.new do |parser|
+      parser.banner = "Usage: ./project.rb #{cmd_name} [options]"
+      parser.on(
+        "--environment ENV", ENV_CHOICES,
+        "Environment (default: local-test): [#{ENV_CHOICES.join(" ")}]") do |v|
+        # The default environment file (called "dev" in Angular language)
+        # compiles a local server to run against the deployed remote test API.
+        self.env = v == "local-test" ? "dev" : v
+      end
+    end
+    parser.parse args
+    self
+  end
+end
+
+# Command line parsing and run logic for deploy-ui command
+class DeployUI
+  attr_reader :common, :opts
+
+  def initialize(cmd_name, args)
+    @common = Common.new
+    @cmd_name = cmd_name
+    @args = args
+    @parser = create_parser(cmd_name)
+    @opts = Options.new
+  end
+
+  def add_options
+    # TODO: Make flag handling more consistent with api/devstart.rb
+    @parser.on("--project [PROJECT]",
+        "Project to create credentials for (e.g. all-of-us-workbench-test). Required.") do |project|
+      @opts.project = project
+    end
+    @parser.on("--account [ACCOUNT]",
+      "Service account to act as for deployment, if any. Defaults to the GAE " +
+      "default service account.") do |account|
+      @opts.account = account
+    end
+    @parser.on("--version [VERSION]",
+          "The name of the version to deploy. Required.") do |version|
+       @opts.version = version
+    end
+    @parser.on("--promote",
+               "Promote this version to immediately begin serving UI traffic. " +
+               "Required: must set --promote or --no-promote") do
+       @opts.promote = true
+    end
+    @parser.on("--no-promote",
+               "Deploy, but do not yet serve traffic from this version. " +
+               "Required: must set --promote or --no-promote") do
+       @opts.promote = false
+    end
+  end
+
+  def validate_options
+    if @opts.project == nil || @opts.version == nil || @opts.promote == nil
+      puts @parser.help
+      exit 1
+    end
+  end
+
+  def run
+    add_options
+    @parser.parse @args
+    validate_options
+    environment_names = {
+      "all-of-us-public-ui-test" => "test",
```

Comment:
This makes it seem like there's going to be a separate project for public-ui.... I don't think that's a good idea. We can deploy as a separate service within the same GCP project, much like we do with api and public-api. So I'd keep these values the way they were for ui.

Reply:
Yeah. I'll fix this. Speaking of that , I tried to make a new service last week in the clound console for public-ui  but I could not. Neil Norman said that I was an owner in the project. How do you make a new service ?

Reply:
You don't make services in cloud console. You just deploy them using gcloud app deploy.

Reply:
Ok. So there is a service item in app.yaml. I set that to public-ui .. Deploy works.